include/circt-c/Dialect/OM.h

```diff
@@ -187,6 +187,22 @@ MLIR_CAPI_EXPORTED bool omEvaluatorValueIsAMap(OMEvaluatorValue evaluatorValue);
 MLIR_CAPI_EXPORTED MlirType
 omEvaluatorMapGetType(OMEvaluatorValue evaluatorValue);
 
+/// Query if the EvaluatorValue is a FrozenBasePath.
+MLIR_CAPI_EXPORTED bool
+omEvaluatorValueIsAFrozenBasePath(OMEvaluatorValue evaluatorValue);
```

Comment:
nit: I think it would be good to keep the C API aligned with the C++ names like we do for the other EvaluatorValue types. So maybe drop the "Frozen" from the name?

Reply:
Or, perhaps it is the C++ names that should be changed, since FrozenBasePathOp leads to BasePath and FrozenPathOp leads to Path.

Reply:
I wanted to call them Path and BasePath originally, but due to the coding patterns ended up with two things called BasePath :).  One of your suggestions below unblocked this, so I have dropped `Frozen` from everywhere.